pkg/cli/alpha/internal/update.go

```diff
@@ -0,0 +1,353 @@
+package internal
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/kubebuilder/v4/pkg/config/store/yaml"
+	"sigs.k8s.io/kubebuilder/v4/pkg/machinery"
+)
+
+// Update contains configuration for the update operation
+type Update struct {
+	// FromVersion specifies which version of Kubebuilder to use for the update.
+	// If empty, the version from the PROJECT file will be used.
+	FromVersion string
+}
+
+// Update performs a complete project update by creating a three-way merge to help users
+// upgrade their Kubebuilder projects. The process creates multiple Git branches:
+// - ancestor: Clean state with old Kubebuilder version scaffolding
+// - current: User's current project state
+// - upgrade: New Kubebuilder version scaffolding
+// - merge: Attempts to merge upgrade changes into current state
+func (opts *Update) Update() error {
+	// Load the PROJECT configuration file to get the current CLI version
+	projectConfigFile := yaml.New(machinery.Filesystem{FS: afero.NewOsFs()})
+	if err := projectConfigFile.LoadFrom(yaml.DefaultPath); err != nil { // TODO: assess if DefaultPath could be renamed to a more self-descriptive name
+		return fmt.Errorf("fail to run command: %w", err)
+	}
+
+	// Determine which Kubebuilder version to use for the update
+	cliVersion := projectConfigFile.Config().GetCliVersion()
```

Comment:
this code is only needed when  opts.FromVersion != "" is nill 
so could we have something like 

defineFrom func where we centralize this logic 
so we will ensure that 

IF FromVersion is not nil then return
ELSE
FromVersion is == PROJECT file value and we use FromVersion in the rest of the logic